src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ParameterCapturingLogger.cs

```diff
@@ -86,6 +88,14 @@ private void ThreadProc()
             }
         }
 
+        public void Complete()
+        {
+            //TODO: Should this also be called during shutdown to ensure all messages are logged?
+
+            _messages.CompleteAdding();
```

Comment:
This is no longer called by product code, so this change as-is seems like it would a regression.

Reply:
The previous Disposal code wasn't getting called at all due to a merge conflict. But putting that aside, the TODO here is meant to address how we want to cancel background logging at the end of the ParameterCapturingService. Do we force logging of all remaining background types (potentially delaying process shutdown to do so), or do we make Disposal immediately stop processing all messages? I'm not sure which direction I wanted to go with it, but want to address the test flakiness first.

Reply:
> The previous Disposal code wasn't getting called at all due to a merge conflict.

Okay, I see that. But it feels like you are just "kicking the can down the road" here.

> But putting that aside, the TODO here is meant to address how we want to cancel background logging at the end of the ParameterCapturingService. Do we force logging of all remaining background types (potentially delaying process shutdown to do so), or do we make Disposal immediately stop processing all messages? I'm not sure which direction I wanted to go with it, but want to address the test flakiness first.

I have no problem with deciding that later.